tools/wpt/browser.py

```diff
@@ -516,264 +516,201 @@ def version(self, binary=None, webdriver_binary=None):
         return None
 
 
-class Chrome(Browser):
-    """Chrome-specific interface.
-
-    Includes webdriver installation, and wptrunner setup methods.
+class ChromeChromiumBase(Browser):
+    """
+    Chromium base class for shared functionality between Chrome and Chromium
     """
 
-    product = "chrome"
     requirements = "requirements_chromium.txt"
-    platforms = {
+    platform = {
         "Linux": "Linux",
         "Windows": "Win",
         "Darwin": "Mac",
-    }
-
-    def __init__(self, logger):
-        super(Chrome, self).__init__(logger)
-        self._last_change = None
-
-    def download(self, dest=None, channel=None, rename=None):
-        if channel != "nightly":
-            raise NotImplementedError("We can only download Chrome Nightly (Chromium ToT) for you.")
-        if dest is None:
-            dest = self._get_dest(None, channel)
-
-        filename = self._chromium_package_name() + ".zip"
-        url = self._latest_chromium_snapshot_url() + filename
-        self.logger.info("Downloading Chrome from %s" % url)
-        resp = get(url)
-        installer_path = os.path.join(dest, filename)
-        with open(installer_path, "wb") as f:
-            f.write(resp.content)
-        return installer_path
-
-    def install(self, dest=None, channel=None):
-        if channel != "nightly":
-            raise NotImplementedError("We can only install Chrome Nightly (Chromium ToT) for you.")
-        dest = self._get_dest(dest, channel)
-
-        installer_path = self.download(dest, channel)
-        with open(installer_path, "rb") as f:
-            unzip(f, dest)
-        os.remove(installer_path)
-        return self.find_nightly_binary(dest)
-
-    def install_mojojs(self, dest, channel, browser_binary):
-        if channel == "nightly" or channel == "canary":
-            url = self._latest_chromium_snapshot_url() + "mojojs.zip"
-        else:
-            chrome_version = self.version(binary=browser_binary)
-            assert chrome_version, "Cannot determine the version of Chrome"
-            # Remove channel suffixes (e.g. " dev").
-            chrome_version = chrome_version.split(' ')[0]
-            url = "https://storage.googleapis.com/chrome-wpt-mojom/%s/linux64/mojojs.zip" % chrome_version
+    }.get(uname[0])
 
-        extracted = os.path.join(dest, "mojojs", "gen")
-        last_url_file = os.path.join(extracted, "DOWNLOADED_FROM")
-        if os.path.exists(last_url_file):
-            with open(last_url_file, "rt") as f:
-                last_url = f.read().strip()
-            if last_url == url:
-                self.logger.info("Mojo bindings already up to date")
-                return extracted
-            rmtree(extracted)
+    # The last revision number used to successfully download/install browser binary or webdriver.
+    last_revision_used = None
```

Comment:
Is this used for Chrome or just Chromium? For Chrome, we can't install the binaries and need to find the driver via https://chromedriver.storage.googleapis.com/LATEST_RELEASE_99 or similar, so I guess just Chromium?

For both this and `revision_attempted`, I wonder if we could keep the information on the file system so that we can check on subsequent runs that we have exactly the binaries that we need?

Reply:
I think that sounds reasonable. Perhaps that work can be in the scope of adding the infrastructure to pin the Chromium revision?